File: java/src/main/java/com/ellen/java/jmvp/di/module/LoginModule.java
package com.ellen.java.jmvp.di.module;

import com.ellen.java.jmvp.login.LoginContract;
import com.ellen.java.jmvp.login.LoginModel;
import com.ellen.java.jmvp.login.LoginPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class LoginModule {

    private LoginContract.LoginView view;

    public LoginModule(LoginContract.LoginView view){
        this.view = view;
    }

    @Provides
    LoginPresenter provideLoginPresenter(LoginModel loginModel, LoginContract.LoginView loginView){
        return new LoginPresenter(loginModel,loginView);
    }

    @Provides
    LoginModel provideLoginModel(){
        return new LoginModel();
    }

    @Provides
    LoginContract.LoginView provideLoginView(){
        return view;
    }

}

File: app/src/main/java/com/ellen/androidarchitecture/di/module/LoginModule.kt
package com.ellen.androidarchitecture.di.module

import com.ellen.androidarchitecture.kmvp.login.LoginContract
import com.ellen.androidarchitecture.kmvp.login.LoginModel
import com.ellen.androidarchitecture.kmvp.login.LoginPresenter
import dagger.Module
import dagger.Provides

@Module
class LoginModule(private var view: LoginContract.LoginView) {

    @Provides
    fun provideLoginPresenter(loginModel: LoginModel,loginView: LoginContract.LoginView) : LoginPresenter{
        return LoginPresenter(loginModel,loginView)
    }

    @Provides
    fun provideLoginModel() : LoginModel{
        return LoginModel()
    }

    @Provides
    fun provideLoginView():LoginContract.LoginView{
        return view
    }
}